docs/configuration/configuration.md

```diff
@@ -104,129 +95,130 @@ Additionally, `relabel_configs` allow advanced modifications to any
-# conflicting labels in the scraped data to "exported_<original-label>" (for
-# example "exported_instance", "exported_job") and then attaching server-side
-# labels. This is useful for use cases such as federation, where all labels
-# specified in the target should be preserved.
-#
-# Note that any globally configured "external_labels" are unaffected by this
-# setting. In communication with external systems, they are always applied only
-# when a time series does not have a given label yet and are ignored otherwise.
-[ honor_labels: <boolean> | default = false ]
-
-# Configures the protocol scheme used for requests.
-[ scheme: <scheme> | default = http ]
-
-# Optional HTTP URL parameters.
-params:
-  [ <string>: [<string>, ...] ]
-
-# Sets the `Authorization` header on every scrape request with the
-# configured username and password.
-basic_auth:
-  [ username: <string> ]
-  [ password: <secret> ]
-
-# Sets the `Authorization` header on every scrape request with
-# the configured bearer token. It is mutually exclusive with `bearer_token_file`.
-[ bearer_token: <secret> ]
-
-# Sets the `Authorization` header on every scrape request with the bearer token
-# read from the configured file. It is mutually exclusive with `bearer_token`.
-[ bearer_token_file: /path/to/bearer/token/file ]
-
-# Configures the scrape request's TLS settings.
-tls_config:
-  [ <tls_config> ]
-
-# Optional proxy URL.
-[ proxy_url: <string> ]
-
-# List of Azure service discovery configurations.
-azure_sd_configs:
-  [ - <azure_sd_config> ... ]
-
-# List of Consul service discovery configurations.
-consul_sd_configs:
-  [ - <consul_sd_config> ... ]
-
-# List of DNS service discovery configurations.
-dns_sd_configs:
-  [ - <dns_sd_config> ... ]
-
-# List of EC2 service discovery configurations.
-ec2_sd_configs:
-  [ - <ec2_sd_config> ... ]
-
-# List of OpenStack service discovery configurations.
-openstack_sd_configs:
-  [ - <openstack_sd_config> ... ]
-
-# List of file service discovery configurations.
-file_sd_configs:
-  [ - <file_sd_config> ... ]
-
-# List of GCE service discovery configurations.
-gce_sd_configs:
-  [ - <gce_sd_config> ... ]
-
-# List of Kubernetes service discovery configurations.
-kubernetes_sd_configs:
-  [ - <kubernetes_sd_config> ... ]
-
-# List of Marathon service discovery configurations.
-marathon_sd_configs:
-  [ - <marathon_sd_config> ... ]
-
-# List of AirBnB's Nerve service discovery configurations.
-nerve_sd_configs:
-  [ - <nerve_sd_config> ... ]
-
-# List of Zookeeper Serverset service discovery configurations.
-serverset_sd_configs:
-  [ - <serverset_sd_config> ... ]
-
-# List of Triton service discovery configurations.
-triton_sd_configs:
-  [ - <triton_sd_config> ... ]
-
-# List of labeled statically configured targets for this job.
-static_configs:
-  [ - <static_config> ... ]
-
-# List of target relabel configurations.
-relabel_configs:
-  [ - <relabel_config> ... ]
+scrape_configs:
```

Comment:
scrape_configs is not a member of scrape_configs

Reply:
It's not? I am unclear what you're saying here?

Reply:
I'm trying to articulate:

```yaml
scrape_configs:
  - job_name: 'prometheus'
    static_configs:
      - targets: ['localhost:9090']
```

Reply:
You're mixing up scrape_config and scrape_configs. What you're proposing removes the definitions of the overall config, scrape_configs, alerting, alertmanager and alerting. I'm open to re-jigging things, but our docs need to stay complete.

In addition examples are not reference documentation, and I'd like to avoid spreading too many example configs about.

Reply:
I am a bit confused then:

1. `scrape_configs` is the top level block right? A `scrap_config` is one or more jobs below that block?

2. This actually *adds* alerting and alertmanager - which previously weren't mentioned.

3. Happy to remove examples but the current page isn't a reference and isn't clear to me because of the inlined configuration. See my discussion below about releveling headers.

Reply:
Yes.

Both alerting and alertmanagers are documented currently. Your proposal does not document alertmanagers or alert_relabel_configs.

In what way do you see the current document as not a reference? As far as I'm aware it correctly mentions all configuration options.

Reply:
Okay - I'm probably not articulating myself well! :) It's not so much what it does or doesn't reference but how.

So I am a newcomer to Prometheus. I download it and open up the default config file and go ... "I wonder what this means?" I go to the configuration page and see the inline section and this is menu:

<img width="315" alt="screenshot 2017-11-21 08 32 53" src="https://user-images.githubusercontent.com/4365/33075200-af02aa96-ce96-11e7-8382-ba846538b7b6.png">

To me a single level of hierarchy menu and the inline section is a disconnect. I get no sense of the configuration file's structure and hierarchy. YMMV naturally - I'll just pushed a menu relevel - not perfect but perhaps better articulates what I was trying to do. If it doesn't work, I am happy to leave it rather than belabor things for folks.

Reply:
I can see what you're trying to do, but you need to do it consistently and not skip over parts of the config. For example why does scrape_config get a separate section but alertmanager doesn't? A bit of inlineing like we currently do helps keep things more succinct.

Reply:
Okay. To me the current menu is far more inconsistent because it skips all the inlined items. But other's mileage varies. :) I'll close this one. Thanks for walking through it!